homeassistant/components/hanna/config_flow.py

```diff
@@ -0,0 +1,101 @@
+"""Config flow for Hanna Instruments integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from hanna_cloud import HannaCloudClient
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_CODE, CONF_EMAIL, CONF_PASSWORD, CONF_SCAN_INTERVAL
+
+from .const import DEFAULT_ENCRYPTION_KEY, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class HannaConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Hanna Instruments."""
+
+    VERSION = 1
+    data_schema = vol.Schema(
+        {
+            vol.Required(CONF_EMAIL): str,
+            vol.Required(CONF_PASSWORD): str,
+            vol.Required(CONF_CODE, default=DEFAULT_ENCRYPTION_KEY): str,
+            vol.Required(CONF_SCAN_INTERVAL, default=5): int,
```

Comment:
We don't allow custom configurable update intervals. People can use an automation for that